### PR TITLE
feat: add LocationSelect component

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes: 1) https://github.com/calcom/cal.com/issues/6394 
2) after clicking edit icon the selected option did not show icon

Screenshot 2023-01-19 at 8 07 54 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by PeerRich** [comment]
nice!

**Comment by PeerRich** [comment]
NIT: small merge conflict 

rest looks good!

CleanShot 2023-01-19 at 15 41 09@2x

**Comment by Udit-takkar** [comment]
should be fixed now

**Comment by sean-brydon** [comment]
Super nice well done - can we possibly have this in storybook?

**Comment by Udit-takkar** [comment]
@sean-brydon i don't think we should add this to storybook because LocationSelect cannot be customized much. it's just that the code was being duplicated so i created a reusable component.

File: apps/web/components/eventtype/EventSetupTab.tsx [hunk 1]
```diff
@@ -17,11 +17,21 @@ import { Button, Icon, Label, Select, SettingsToggle, Skeleton, TextField } from
...
+import LocationSelect, {
+  SingleValueLocationOption,
+  LocationOption,
+} from "@components/ui/form/LocationSelect";
 
-type OptionTypeBase = {
-  label: string;
-  value: EventLocationType["type"];
-  disabled?: boolean;
+const getLocationFromType = (
```
Comment (emrysal) [inline apps/web/components/eventtype/EventSetupTab.tsx:25]: I committed in a small stylistic change for this function, functionally the same though 👍

File: apps/web/components/dialog/EditLocationDialog.tsx [hunk 2]
```diff
@@ -211,7 +203,7 @@ export const EditLocationDialog = (props: ISetLocationDialog) => {
   })();
 
   return (
-    <Dialog open={isOpenDialog}>
+    <Dialog open={isOpenDialog} onOpenChange={(open) => setShowLocationModal(open)}>
```
Comment (emrysal) [inline apps/web/components/dialog/EditLocationDialog.tsx:206]: Fixed a NIT that prevented the dialog to close when you clicked outside.

**Review by emrysal — APPROVED** [review]
Excellent work on the custom select!

**Pull request merged**